src/NossoMercadoLivreAPI.Infra.Data/Repository/UserRepository.cs

```diff
@@ -10,16 +11,16 @@ namespace NossoMercadoLivreAPI.Infra.Data.Repository
 {
     public class UserRepository : IUserRepository
     {
-        private readonly ContextDb _context;
-        public UserRepository(ContextDb context)
+        private readonly IDbContext _context;
+        public UserRepository(IDbContext context)
         {
             _context = context;
         }
 
         public virtual async Task<User> InsertAsync(User entity)
         {
-            await _context.Set<User>().AddAsync(entity);
-            await _context.SaveChangesAsync();
+            await _context.Instance.Set<User>().AddAsync(entity);
+            await _context.Instance.SaveChangesAsync();
```

Comment:
*Só alteramos estado de referências que criamos. Não mexemos nos objetos alheios. A não ser que esse objeto seja criado para isso, como é o caso de argumentos de métodos de borda mais externa. Estes são, geralmente, associados a frameworks.*

Não existe outro jeito de você controlar o commit?

Reply:
No momento a única forma que eu vejo para controlar o commit é chamando o ContextDb direto, do jeito que era antes. Pela interface que eu adicionei foi necessário criar uma instancia para ter acesso a todas funcionalidades do entity framework. No ContextDb essa instancia da interface é injetada desse modo "public DbContext Instance => this;" e assim a usabilidade fica do mesmo jeito que a versão anterior, mas com a adição desse instance ao usar a interface.